driver-core/src/test/java/com/datastax/driver/core/CCMTestsSupport.java

```diff
@@ -0,0 +1,687 @@
+
+    private TestMode testMode;
+
+    private CCMTestConfig ccmTestConfig;
+
+    private CCMTestContext ccmTestContext;
+
+    protected Cluster cluster;
+
+    protected Session session;
+
+    protected String keyspace;
+
+    @BeforeClass(groups = {"isolated", "short", "long"})
+    public void setUp() throws Exception {
+        setUp(this);
+    }
+
+    public void setUp(Object testInstance) throws Exception {
+        this.testMode = determineTestMode(testInstance.getClass());
+        if (testMode == PER_CLASS)
+            initTestContext(testInstance, null);
+    }
+
+    @BeforeMethod(groups = {"isolated", "short", "long"})
+    public void setUp(Method testMethod) throws Exception {
+        setUp(this, testMethod);
+    }
+
+    public void setUp(Object testInstance, Method testMethod) throws Exception {
+        if (testMode == PER_METHOD)
+            initTestContext(testInstance, testMethod);
+    }
+
+    @AfterMethod(groups = {"isolated", "short", "long"}, alwaysRun = true)
+    public void tearDown() {
+        if (testMode == PER_METHOD)
+            closeTestContext();
+    }
+
+    @AfterClass(groups = {"isolated", "short", "long"}, alwaysRun = true)
+    public void cleanUp() {
+        if (testMode == PER_CLASS)
+            closeTestContext();
+    }
+
+
+    /**
+     * Returns the CCMBridge builder to use for this test.
+     * <p/>
+     * The default implementation returns a standard CCM cluster configuration;
+     * user-defined functions are enabled for protocol versions >= 4.
+     *
+     * @return The CCM config to use for the tests.
+     */
+    public CCMBridge.Builder createCCMBridgeBuilder() {
+        CCMBridge.Builder builder = CCMBridge.builder()
+                .withoutNodes()
+                .notStarted();
+        ProtocolVersion protocolVersion = TestUtils.getDesiredProtocolVersion();
+        if (protocolVersion.toInt() >= 4) {
+            builder = builder.withCassandraConfiguration("enable_user_defined_functions", "true");
+        }
+        return builder;
+    }
+
+    /**
+     * Returns the cluster builder to use for this test.
+     * <p/>
+     * The default implementation returns a vanilla builder.
+     *
+     * It's not required to call {@link com.datastax.driver.core.Cluster.Builder#addContactPointsWithPorts},
+     * it will be done automatically.
+     *
+     * @return The cluster builder to use for the tests.
+     */
+    public Cluster.Builder createClusterBuilder() {
+        return Cluster.builder();
+    }
+
+    /**
+     * Returns an alternate cluster builder to use for this test,
+     * with no event debouncing.
+     * <p/>
+     * It's not required to call {@link com.datastax.driver.core.Cluster.Builder#addContactPointsWithPorts},
+     * it will be done automatically.
+     *
+     * @return The cluster builder to use for the tests.
+     */
+    @SuppressWarnings("unused")
+    public Cluster.Builder createClusterBuilderNoDebouncing() {
```

Comment:
Yes! I use it with CaseSensitivityTest, but other tests can definitely use this alternate builder.